go/libraries/doltcore/doltdb/branch_activity.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package doltdb
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+const (
+	READ = iota
+	WRITE
+)
+
+type statsSessionContextKeyType struct{}
+
+// StatsSessionContextKey is used to mark sql sessions which are related to stats processing. We don't want to count
+// reads/writes from these sessions in branch activity tracking.
+var StatsSessionContextKey = statsSessionContextKeyType{}
```

Comment:
Nit: I think it's slightly atypical to expose these. Better is just

```
func WithAsStatsSessionContext(ctx context.Context) context.Context
```

or something like that. This is probably more of a concern when people are putting actual data in them that this code is then going to use / pull out, but even in this case it seems better encapsulation and readability.